### ISSUE TITLE
Connectivity with VPN and proxy broken - no such host

### ISSUE THREAD

**Issue by bugra-derre** [docker/for-mac#6315]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 872096FA-4E99-4B73-8D7E-A05ED61B8E77/20220512100418

### Expected behavior
Docker respects the proxy settings so it can resolve domains and can download container images

### Actual behavior
Docker responds
> docker: Error response from daemon: Get "https://registry-1.docker.io/v2/": Failed to lookup host: registry-1.docker.io.

### Information

  - macOS Version:
  - Intel chip or Apple chip: Apple chip
  - Docker Desktop Version:
     

> Client:
>  Cloud integration: v1.0.24
>  Version:           20.10.14
>  API version:       1.41
>  Go version:        go1.16.15
>  Git commit:        a224086
>  Built:             Thu Mar 24 01:49:20 2022
>  OS/Arch:           darwin/arm64
>  Context:           default
>  Experimental:      true
> 
> Server: Docker Desktop 4.8.1 (78998)
>  Engine:
>   Version:          20.10.14
>   API version:      1.41 (minimum version 1.12)
>   Go version:       go1.16.15
>   Git commit:       87a90dc
>   Built:            Thu Mar 24 01:45:44 2022
>   OS/Arch:          linux/arm64
>   Experimental:     false
>  containerd:
>   Version:          1.5.11
>   GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
>  runc:
>   Version:          1.0.3
>   GitCommit:        v1.0.3-0-gf46b6ba
>  docker-init:
>   Version:          0.19.0
>   GitCommit:        de40ad0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

> [PASS] DD0027: is there available disk space on the host?
> [PASS] DD0028: is there available VM disk space?
> [PASS] DD0031: does the Docker API work?
> [PASS] DD0004: is the Docker engine running?
> [PASS] DD0011: are the LinuxKit services running?
> [PASS] DD0016: is the LinuxKit VM running?
> [PASS] DD0001: is the application running?
> [PASS] DD0018: does the host support virtualization?
> [PASS] DD0017: can a VM be started?
> [PASS] DD0015: are the binary symlinks installed?
> [PASS] DD0003: is the Docker CLI working?
> [PASS] DD0013: is the $PATH ok?
> [PASS] DD0007: is the backend responding?
> [PASS] DD0014: are the backend processes running?
> [PASS] DD0008: is the native API responding?
> [PASS] DD0009: is the vpnkit API responding?
> [PASS] DD0010: is the Docker API proxy responding?
> [FAIL] DD0012: is the VM networking working? network checks failed: failed to lookup hub.docker.com: lookup hub.docker.com on 192.168.65.5:53: no such host
> [2022-05-12T08:49:28.982829000Z][com.docker.diagnose][I] ipc.NewClient: 4402a718-diagnose-network -> Library/Containers/com.docker.docker/Data/diagnosticd.sock diagnosticsd
> [common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
> [	common/pkg/diagkit/gather/diagnose/network.go:34 +0xd4
> [common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x101c500e0)
> [	common/pkg/diagkit/gather/diagnose/test.go:46 +0x44
> [common/pkg/diagkit/gather/diagnose.Run.func1(0x101c500e0)
> [	common/pkg/diagkit/gather/diagnose/run.go:17 +0x44
> [common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2?, 0x101c500e0)
> [	common/pkg/diagkit/gather/diagnose/run.go:140 +0x84
> [common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x101c500e0, 0x1400069d728)
> [	common/pkg/diagkit/gather/diagnose/run.go:146 +0x3c
> [common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x20?, 0x1400069d728)
> [	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
> [common/pkg/diagkit/gather/diagnose.walkOnce(0x1017340a0?, 0x140002df890)
> [	common/pkg/diagkit/gather/diagnose/run.go:135 +0x8c
> [common/pkg/diagkit/gather/diagnose.Run(0x101c50260, 0x101003fa0?, {0x140002dfb08, 0x1, 0x1})
> [	common/pkg/diagkit/gather/diagnose/run.go:16 +0x178
> [main.checkCmd({0x14000032050?, 0x6?, 0x4?}, {0x0, 0x0})
> [	common/cmd/com.docker.diagnose/main.go:132 +0xe0
> [main.main()
> [	common/cmd/com.docker.diagnose/main.go:98 +0x308
> [2022-05-12T08:49:28.982906000Z][com.docker.diagnose][I] (23de1b0c) 4402a718-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["192.168.2.137","10.141.159.101"]}
> [2022-05-12T08:49:29.069229000Z][com.docker.diagnose][W] (23de1b0c) 4402a718-diagnose-network C<-S e171ba9b-diagnosticsd POST /check-network-connectivity (86.332375ms): failed to lookup hub.docker.com: lookup hub.docker.com on 192.168.65.5:53: no such host
> 
> [PASS] DD0032: do Docker networks overlap with host IPs?
> [SKIP] DD0030: is the image access management authorized?
> [PASS] DD0019: is the com.docker.vmnetd process responding?
> [PASS] DD0033: does the host have Internet access?

### Steps to reproduce the behavior

  1. Sitting together with colleagues having a Mac & Docker Desktop. Each facing the same proxy.
  2. docker run hello-world runs for them, but not for me
  3. Proxy settings are all the same for colleagues & me. Only difference is the used Docker Desktop version (colleagues have older Docker Desktop for Mac versions).
  4. Side note: When connected to the "real internet" docker works like a charm, but: the proxy settings are not unset (they are still enriched in the environment + the Docker settings). For me it seems, as if the proxy settings are not injected to Docker.

**Comment by bugra-derre** [comment]
To add on this. I installed an older version for Docker for Mac (4.6.1). Everything works as expected. I guess sth is wrong with the current release.

**Comment by francois-gsk** [comment]
> Thank you @bugra-derre for the report. I think we've identified the problem in 4.8.1 and I can share an internal build with you. The manual proxy settings should now work again.
>     * [Mac Intel](https://desktop-stage.docker.com/mac/main/amd64/79160/Docker.dmg)

@StefanScherer for the __Intel version__ it's even worst for our use case. In 4.8.1, I was able to pull images within our network but not thru the proxy, but with this version I can't pull any images:

```
Error response from daemon: Get "https://[REDACTED]": Proxy Authentication Required
```

But I'm authenticated with the proxy.

So:
- `build 79160`, `4.8.1`, `4.8.0`: proxy issues
- `4.7.1`: it's 👌🏼

**Comment by laurazard** [comment]
Hi! Thank you both for the information and the reports, it's really useful for us to figure out the issue! If you feel like it, could you test another internal build?
- [Mac Intel](https://desktop-stage.docker.com/mac/main/amd64/79243/Docker.dmg)
- [Mac M1](https://desktop-stage.docker.com/mac/main/arm64/79243/Docker.dmg)
- [Windows](https://desktop-stage.docker.com/win/main/amd64/79243/Docker%20Desktop%20Installer.exe)
I believe the issue with proxy authentication is fixed in this build, but I'd love your feedback @francois-gsk

**Comment by francois-gsk** [comment]
> Hi! Thank you both for the information and the reports, it's really useful for us to figure out the issue! If you feel like it, could you test another internal build?
>     * [Mac Intel](https://desktop-stage.docker.com/mac/main/amd64/79243/Docker.dmg)
> I believe the issue with proxy authentication is fixed in this build, but I'd love your feedback @francois-gsk

Will do ASAP! But first finalizing some PR 😅

**Comment by francois-gsk** [comment]
@laurazard & @StefanScherer 
Fixed! Version `4.9.0` (build 79243) is working 👌🏼 on __Mac Intel__

**Comment by andrianaivo** [comment]
> Fixed! Version 4.9.0 (build 79243) is working 👌🏼 on Mac Intel

Yes, `4.9.0` also fixed my issue on **Mac M1**

**Comment by ismailsalloo** [comment]
`4.9.0` also fixed my Mac M1 issue, but I have to manually put the proxy details (I use CNTLM for my corporate proxy). Is that the desired intention or should it have picked up the system proxy settings (which is a pac file)?

**Comment by rumstead** [comment]
4.8.1 breaks proxies and 4.9.0 fixes them. Using MAC Intel.

**Comment by JodieHag** [comment]
Where is the 4.9.0??

**Comment by rumstead** [comment]
> Hi! Thank you both for the information and the reports, it's really useful for us to figure out the issue! If you feel like it, could you test another internal build?
> 
> * [Mac Intel](https://desktop-stage.docker.com/mac/main/amd64/79243/Docker.dmg)
> * [Mac M1](https://desktop-stage.docker.com/mac/main/arm64/79243/Docker.dmg)
> * [Windows](https://desktop-stage.docker.com/win/main/amd64/79243/Docker%20Desktop%20Installer.exe)
> 
> I believe the issue with proxy authentication is fixed in this build, but I'd love your feedback @francois-gsk

@JodieHag this comment has the links

**Comment by JodieHag** [comment]
Thanks, it's amazing. Works for me!

**Comment by sagarraje** [comment]
I have installed  [Mac M1](https://desktop-stage.docker.com/mac/main/arm64/79243/Docker.dmg) and its working fine :)

**Comment by mat007** [comment]
Hi, we have just released Docker Desktop 4.8.2 with a fix. Thanks!
